### PR TITLE
Ryu Flow Mod v1.3 Implemented (D113)

### PR DESCRIPTION
**Pull request by steven19920206**
1. Fixed error in repack_flow_13
2. Add 'to_match' function -- convert match key from OmniUI to ryu
3. Add 'to_action' function -- handle mutiple actions from OmniUI
4. Update README.md -- add Ryu 1.3 Flow Mod Test Environment

### CODE REVIEW

**Comment by steven19920206** [comment]
1. fixed error in repack_flow_13
2. add 'to_match' function -- convert match key from OmniUI to ryu
3. add 'to_action' function -- handle mutiple actions from OmniUI
4. Update README.md -- add Ryu 1.3 Flow Mod Test Environment

File: adapter/ryu/README.md [hunk 1]
```diff
@@ -21,3 +21,18 @@ $ `sudo mn --topo single,3 --mac --switch ovsk --controller remote`
 4. Start-up OmniUI Web UI (Using a 4th Terminal)  
 $ `firefox ~/OmniUI/webui/index.html`
 
+###How to execution Ryu 1.3###
```
Comment (rascov) [inline adapter/ryu/README.md]: Make the instruction more consistent with the previous instruction

File: adapter/ryu/omniui/omniui.py [hunk 2]
```diff
@@ -198,13 +204,235 @@ def links(self, req, **kwargs):
             reverse = False
             for link in result:
                 if(link['src-switch'] == omniLink['dst-switch'] and
-										link['dst-switch'] == omniLink['src-switch'] and
-										link['src-port'] == omniLink['dst-port'] and
-										link['dst-port'] == omniLink['src-port']):
+                                        link['dst-switch'] == omniLink['src-switch'] and
```
Comment (rascov) [inline adapter/ryu/omniui/omniui.py]: Unnecessary indent spaces?

File: adapter/ryu/omniui/omniui.py [hunk 3]
```diff
@@ -198,13 +204,235 @@ def links(self, req, **kwargs):
...
+        elif cmd == 'DEL':
+            cmd = dp.ofproto.OFPFC_DELETE
+        elif cmd == 'DEL_ST':
+            cmd = dp.ofproto.OFPFC_DELETE_STRICT
+        else:
+            return Response(status=404)
+            
+        # repack Omniui Flow to Ryu Flow
+        if dp.ofproto.OFP_VERSION == ofproto_v1_0.OFP_VERSION:
+            ofctl_v1_0.mod_flow_entry(dp, ryuFlow, cmd)
```
Comment (rascov) [inline adapter/ryu/omniui/omniui.py]: ryuFlow is not defined in this if clause

**Comment by rascov** [comment]
By the way, please resolve the conflicts. Thanks.

**Comment by steven19920206** [comment]
merge OF1.0 and OF1.3

File: adapter/ryu/omniui/omniui.py [hunk 4]
```diff
@@ -212,25 +212,24 @@ def links(self, req, **kwargs):
...
         try:
-            omniFlow = ast.literal_eval(req.body)     #Getting flow from req
+            omniFlow = ast.literal_eval(req.body)
         except SyntaxError:
-            LOG.debug('Invalid syntax %s', req.body)
+            LOG.debug('invalid syntax %s', req.body)
             return Response(status=400)
 
-        omniDpid = omniFlow.get('switch')             #Getting OmniUI dpid from flow    
-        if omniDpid is None:
```
Comment (rascov) [inline adapter/ryu/omniui/omniui.py]: Why removing the null checker?

**Comment by steven19920206** [comment]
add null checker for dpid

**Comment by rascov** [comment]
Manually merged 5937ec7

**Pull request closed without merging**